modules/ld/sparql-client/src/main/scala/ch/epfl/bluebrain/nexus/commons/sparql/client/SparqlClient.scala

```diff
@@ -116,6 +116,28 @@ class SparqlClient[F[_]](sparqlBase: Uri)(implicit
   }
 
   /**
+    * Checks if a index exists as a namespace
+    *
+    * @param name the name of the index
+    * @return ''Left(())'' when the index does not exist, ''Right(())'' when it does exist
+    *         and it signals an error otherwise.
+    */
+  def indexExists(name: String): F[Either[Unit, Unit]] = {
+    cl(Get(s"$sparqlBase/namespace/$name")).flatMap { resp =>
+      val discardedEntity = cl.discardBytes(resp.entity)
```

Comment:
I would recommend doing something like this such that you also have the opportunity to log the response:
```
resp.status match {
  case StatusCodes.OK =>
    cl.discardBytes(resp.entity)
    F.pure(true)
  case StatusCodes.NotFound =>
    cl.discardBytes(resp.entity)
    F.pure(false)
  case other =>
    cl.toString(resp.entity).flatMap { body =>
          log.error(s"Unexpected Sparql response for verifying the existence of a namespace:\nRequest: '${req.method} ${req.uri}'\nStatus: '${resp.status}'\nResponse: '$body'") // requires extracting the request in a variable 'req'
          F.raiseError(UnexpectedSparqlResponse(resp.status, body))
        }
}
```